File: src/leetcode/p2079/java/Solution.java
package leetcode.p2079.java;

public class Solution {
    public int wateringPlants(int[] plants, int capacity) {
        int answer = 0;
        int water = capacity;
        for (int i = 0; i < plants.length; i++) {
            if (water >= plants[i]) {
                water -= plants[i];
                answer++;
            } else {
                water = capacity - plants[i];
                answer += (i * 2)  + 1;
            }
        }
        return answer;
    }
}

File: src/leetcode/p2079/kotlin/Solution.kt
package leetcode.p2079.kotlin

class Solution {
    fun wateringPlants(plants: IntArray, capacity: Int): Int {
        var answer = 0
        var water = capacity

        plants.forEachIndexed { index, plant ->
            if (water >= plant) {
                water -= plant
                answer++
            } else {
                water = capacity - plant
                answer += index * 2 + 1
            }
        }
        return answer
    }
}